src/main/java/com/example/springTask/services/TeacherService.java

```diff
@@ -20,10 +20,9 @@
     @Autowired
     private TeacherRepository teacherRepository;
 
-    public List<Teacher> getAll(Integer pageNo, Integer pageSize, String sortBy) {
-        Pageable paging = PageRequest.of(pageNo, pageSize, Sort.by(sortBy));
+    public List<Teacher> getAll(Pageable pageable) {
```

Comment:
you have to return an instance of Page object because only in this way frontend will know how many pages are available.